x/logic/predicate/bank_test.go

```diff
@@ -148,7 +167,7 @@ func TestBank(t *testing.T) {
 				balances:   []bank.Balance{},
 				query:      `bank_balances('foo', X).`,
 				wantResult: []testutil.TermResults{{"X": "[uaxone-100]"}},
-				wantError: fmt.Errorf("error(resource_error(resource_module(bank)),[%s],bank_balances/2)",
+				wantError: fmt.Errorf("error(domain_error(encoding(bech32),foo),[%s],bank_balances/2)",
 					strings.Join(strings.Split("decoding bech32 failed: invalid bech32 string length 3", ""), ",")),
 			},
```

Comment:
Alright, I know we’ve discussed this before, and I get the sense you’re not a fan of this approach 😅 (which is understandable). From what I remember, this transformation was initially implemented because error messages used to be atoms, but now they’re strings (essentially a list of characters). The change was made to update the tests without too much hassle.

That said, I agree it would be better to stick with the string as it is, without any transformation. However, I don’t want to address this here since it touches on parts of the code that fall outside the scope of this PR. So, could you create a GitHub issue for us to handle later?